addons/point_of_sale/static/tests/unit/related_models_test.js

```diff
@@ -0,0 +1,948 @@
+            const c1 = models["product.category"].create({});
+            const c2 = models["product.category"].create({});
+            const c3 = models["product.category"].create({ parent_id: c1 });
+
+            c3.update({ parent_id: c2 });
+            assert.equal(c3.parent_id, c2, "Category should be updated with new parent");
+            assert.ok(c2.child_ids.includes(c3), "Category should be linked to the new parent");
+            assert.notOk(
+                c1.child_ids.includes(c3),
+                "Category should be unlinked from the old parent"
+            );
+        });
+
+        QUnit.test("update operation, one2many", (assert) => {
+            const c1 = models["product.category"].create({});
+            const c2 = models["product.category"].create({});
+
+            c1.update({ child_ids: [["link", c2]] });
+            assert.ok(c1.child_ids.includes(c2), "Category should be linked to the parent");
+            assert.equal(c2.parent_id, c1, "Category should be linked to the parent");
+        });
+
+        QUnit.test("update operation, unlink many2one", (assert) => {
+            const c1 = models["product.category"].create({});
+            const c2 = models["product.category"].create({});
+            c2.update({ parent_id: c1 });
+            assert.ok(c2.parent_id, "Category should be linked to the parent");
+
+            c2.update({ parent_id: undefined });
+            assert.notOk(c2.parent_id, "Category should be unlinked from the parent");
+            assert.notOk(c1.child_ids.includes(c2), "Category should be unlinked from the parent");
+        });
+
+        QUnit.test("update operation, unlink one2many", (assert) => {
+            const c1 = models["product.category"].create({});
+            const c2 = models["product.category"].create({});
+            c1.update({ child_ids: [["link", c2]] });
+            assert.ok(c1.child_ids.includes(c2), "Category should be linked to the parent");
+
+            c1.update({ child_ids: [["unlink", c2]] });
+            assert.notOk(c1.child_ids.includes(c2), "Category should be unlinked from the parent");
+            assert.notOk(c2.parent_id, "Category should be unlinked from the parent");
+        });
+
+        QUnit.test("update operation, Clear one2many", (assert) => {
+            const category = models["product.category"].create({ name: "Category" });
+            models["product.category"].create({ name: "Category 1", parent_id: category });
+            models["product.category"].create({ name: "category 2", parent_id: category });
+            models["product.category"].update(category, { child_ids: [["clear"]] });
+            const updatedCategory = models["product.category"].read(category.id);
+            assert.equal(
+                updatedCategory.child_ids.length,
+                0,
+                "All child categories should be unlinked from the category"
+            );
```

Comment:
```suggestion
            assert.equal(
                category.child_ids.length,
                0,
                "All child categories should be unlinked from the category"
            );
```
no need for getting it. original object has the changes already